File: nagasena/src/org/openexi/proc/events/EXIEventWildcardStartElement.java
package org.openexi.proc.events;

import org.openexi.proc.common.BinaryDataSource;
import org.openexi.proc.common.EventType;
import org.openexi.proc.common.EventDescription;
import org.openexi.schema.Characters;

public final class EXIEventWildcardStartElement implements EventDescription {

  private final String m_uri;
  private final String m_name;
  private final int m_uriId;
  private final int m_nameId;
  private final String m_prefix;
  
  private final EventType m_eventType;
  
  public EXIEventWildcardStartElement(String uri, String name, int uriId, int nameId, String prefix, EventType eventType) {
    assert eventType.itemType == EventType.ITEM_SCHEMA_WC_ANY || eventType.itemType == EventType.ITEM_SCHEMA_WC_NS || eventType.itemType == EventType.ITEM_SE_WC; 
    m_uri = uri; 
    m_name = name;
    m_uriId = uriId;
    m_nameId = nameId;
    m_prefix = prefix;
    m_eventType = eventType;
  }
  
  public byte getEventKind() {
    return EventDescription.EVENT_SE;
  }

  public String getName() {
    return m_name;
  }

  public String getURI() {
    return m_uri;
  }

  public int getNameId() {
    return m_nameId;
  }

  public int getURIId() {
    return m_uriId;
  }

  public String getPrefix() {
    return m_prefix;
  }

  public Characters getCharacters() {
    return null;
  }
  
  public BinaryDataSource getBinaryDataSource() {
    return null;
  }
  
  public EventType getEventType() {
    return m_eventType;
  }
  
}

File: nagasenacs/src/Nagasena/Proc/Events/EXIEventWildcardStartElement.cs
using System.Diagnostics;

using BinaryDataSource = Nagasena.Proc.Common.BinaryDataSource;
using EventType = Nagasena.Proc.Common.EventType;
using EventDescription = Nagasena.Proc.Common.EventDescription;
using EventDescription_Fields = Nagasena.Proc.Common.EventDescription_Fields;
using Characters = Nagasena.Schema.Characters;

namespace Nagasena.Proc.Events {

  /// <exclude/>
  public sealed class EXIEventWildcardStartElement : EventDescription {

    private readonly string m_uri;
    private readonly string m_name;
    private readonly int m_uriId;
    private readonly int m_nameId;
    private readonly string m_prefix;

    private readonly EventType m_eventType;

    public EXIEventWildcardStartElement(string uri, string name, int uriId, int nameId, string prefix, EventType eventType) {
      Debug.Assert(eventType.itemType == EventType.ITEM_SCHEMA_WC_ANY || eventType.itemType == EventType.ITEM_SCHEMA_WC_NS || eventType.itemType == EventType.ITEM_SE_WC);
      m_uri = uri;
      m_name = name;
      m_uriId = uriId;
      m_nameId = nameId;
      m_prefix = prefix;
      m_eventType = eventType;
    }

    public sbyte EventKind {
      get {
        return EventDescription_Fields.EVENT_SE;
      }
    }

    public string Name {
      get {
        return m_name;
      }
    }

    public string URI {
      get {
        return m_uri;
      }
    }

    public int NameId {
      get {
        return m_nameId;
      }
    }

    public int URIId {
      get {
        return m_uriId;
      }
    }

    public string Prefix {
      get {
        return m_prefix;
      }
    }

    public Characters Characters {
      get {
        return null;
      }
    }

    public BinaryDataSource BinaryDataSource {
      get {
        return null;
      }
    }

    public EventType getEventType() {
      return m_eventType;
    }

  }

}